socks/socks4.py

```diff
@@ -39,10 +49,17 @@ def dumps(self) -> bytes:
 class SOCKS4Reply(typing.NamedTuple):
     reply_code: SOCKS4ReplyCode
     port: int
-    addr: bytes
+    addr: str
 
-    def loads(self, data: bytes) -> "SOCKS4Reply":
-        raise NotImplementedError()
+    @classmethod
+    def loads(cls, data: bytes) -> "SOCKS4Reply":
+        if len(data) != 8:
+            raise ProtocolError("Malformed reply")
+        return cls(
+            reply_code=SOCKS4ReplyCode(data[1:2]),
```

Comment:
What exception gets raised if we get an invalid reply_code? We should catch and re-raise a ProtocolError for a malformed reply